presto-spi/src/main/java/io/prestosql/spi/Plugin.java

```diff
@@ -30,6 +30,8 @@
 
 public interface Plugin
 {
+    default void setPrestoInformation(PrestoInformation prestoInformation) {}
```

Comment:
I don't like `set` because this implies implementation is mutable. 
If a plugin wants to use this information in some further place than directly in the class implementing `Plugin`, this further complicates implementation. 

I would look into providing this information during Plugin instantiation. For example, Plugin could optionally have a constructor taking `PrestoInformation` 

I would like to understand what are you trying to achieve. In fact, presto version information is something known statically, often this is part of the file system path where the presto (and plugin) is installed. What problem are you solving here? What are alternative approaches?

Reply:
> For example, Plugin could optionally have a constructor taking PrestoInformation

This actually won't work, since `Plugin` is created via `ServiceLoader` which only supports zero argument constructors. We would need a `PluginFactory` service. But that seems a bit silly as `Plugin` itself is already a factory (that is usually providing factories...).